src/lavatory/utils/artifactory.py

```diff
@@ -1,100 +1,90 @@
+    def _parse_artifact_name(name):
+        simple_name = '/'.join(name.split('/')[-4:])
+        return simple_name
 
-        if response.ok:
-            return response.json()
-        else:
-            response.raise_for_status()
+    def list(self):
+        """
+        Return a list of repos with basic info about each
+        If the optional parameter repo_name is specified, then only return
+        information pertaining to that repo
+        """
 
-    def post(self, endpoint, payload):
-        response = requests.post(self.apiurl+endpoint, auth=self.auth, data=payload)
+        repos = {}
 
-        if response.ok:
-            return response.json()
-        else:
-            response.raise_for_status()
+        raw_data = self.artifactory.request('storageinfo')
+        data = raw_data.json()
+        for repo in data["repositoriesSummaryList"]:
+            if repo["repoKey"] == "TOTAL":
+                continue
 
-    def delete(self, path):
-        LOG.info("DELETE", path)
-        return
+            if not self.repo_name or self.repo_name == repo["repoKey"]:
+                repos[repo["repoKey"]] = repo
 
-        """
-        response = requests.delete(baseurl + "/" + path)
+        return repos
 
-        if response.ok:
-            return response.json()
-        else:
-            response.raise_for_status()
-            #raise Exception("HTTP DELETE {}: Server code {}".format(endpoint, response.status_code))
-        """
+    def all_artifacts(self, search='', repo='', depth=3):
+        """ Returns a dict of artifact and properties """
+        LOG.debug('Finding all artifacts with: search=%s, repo=%s, depth=%s', search, repo, depth)
+        all_artifacts = {}
 
+        self.artifactory.find_by_pattern(filename=search, specific_repo=repo, max_depth=depth)
 
-class Artifactory(object):
+        for artifact in sorted(self.artifactory.files):
+            artifact_simple_name = self._parse_artifact_name(artifact)
+            LOG.debug('Found: %s', artifact_simple_name)
+            self.artifactory.get_properties(artifact)
 
-    def __init__(self, baseurl, user, passwd):
-        self.http = _Http(baseurl, user, passwd)
+        LOG.info('Found %d artifacts in total.', len(all_artifacts))
+        return all_artifacts
 
-    def list(self, reponame=None):
-        """
-        Return a list of repos with basic info about each
-        If the optional parameter reponame is specified, then only return
-        information pertaining to that repo
-        """
-    
-        repos = {}
-    
-        json = self.http.get("/storageinfo")
-        for repo in json["repositoriesSummaryList"]:
-            if repo["repoKey"] == "TOTAL" :
-                continue
-                
-            if not reponame or reponame == repo["repoKey"]:
-                repos[repo["repoKey"]] = repo
-    
-        return repos
-    
-    def purge(self, repo, dryrun, artifacts):
+    def purge(self, repo, dry_run, artifacts):
```

Comment:
repo can be removed from this function and use`self.repo_name`

Reply:
changed to repo_name from init.